packages/peer-exchange/src/waku_peer_exchange_discovery.ts

```diff
@@ -55,10 +56,13 @@ export class PeerExchangeDiscovery
     if (!protocols.includes(PeerExchangeCodec) || this.intervals.get(peerId))
       return;
 
-    await this.query(peerId);
+    setTimeout(async () => {
+      await this.query(peerId);
+    }, 5000);
     const interval = setInterval(async () => {
+      this.queryAttempt++;
```

Comment:
I think we should stop increasing it at some point, not sure what the upper boundary should look like.

Reply:
Agreed.

I would instead just give up after X attempts (let's say 3) have failed.

I'd also extract the back off logic in a dedicated function for less mind load.

Probably have a function that handles:
- timeout of querying
- backoff interval
- max retry

Reply:
Agreed as well - addressed